hack/e2e.sh

```diff
@@ -132,6 +132,15 @@ echo "DOCKER_IO_MIRROR: $DOCKER_IO_MIRROR"
 echo "GCR_IO_MIRROR: $GCR_IO_MIRROR"
 echo "QUAY_IO_MIRROR: $QUAY_IO_MIRROR"
 
+# check bash version
+BASH_MAJOR_VERSION=$(echo "$BASH_VERSION" | grep -o "^\d\{1,3\}")
```

Comment:
```suggestion
BASH_MAJOR_VERSION=$(echo "$BASH_VERSION" | cut -d '.' -f 1)
```

Reply:
because it's pre-check of the script, can you move this check after the `source "${ROOT}/hack/lib.sh"`? 
no need to run any logic if the bash is not the version after 4.

Reply:
@cofyc even `./hack/e2e.sh --help`?

Reply:
it makes sense to allow `./hack/e2e.sh -h`, you can update #1565 

by the way, normally you don't need to create another PR to submit new changes.  you can push the branch again to update the PR (add `-f` flag to force it if necessary or pull again to merge remote change)